src/crustls.h

```diff
@@ -281,6 +281,40 @@ typedef struct rustls_verify_server_cert_params {
 
 typedef enum rustls_result (*rustls_verify_server_cert_callback)(rustls_verify_server_cert_user_data userdata, const struct rustls_verify_server_cert_params *params);
 
+/**
+ * A return value for a function that may return either success (0) or an errno
+ * value.
+ */
+typedef int rustls_io_error;
```

Comment:
nit: `rustls_io_result` for consistency with `enum rustls_result`? Also what range of errnos will the nonzero values be drawn from on Windows?